### DISCUSSION TITLE
State Color INSTEAD of Icon Color?

### DISCUSSION THREAD

**Discussion by itsnetservices** [marcokreeft87/room-card#178] [Q&A] [answered]
Thanks so much for creating this amazing card! It's definitely going to improve my dashboard.

Quick question though, is there a way to change the color of a STATE instead of an ICON?

Example, I would like to change the color of the state of my temperature sensors in the info_entities area...

![image](https://user-images.githubusercontent.com/102342439/221889716-dc57ed14-9953-47f7-82df-1a60a23b9c9b.png)

**Comment by marcokreeft87** [comment, marked as answer]
You could obtain your goal with styles : https://github.com/marcokreeft87/room-card/wiki/Configuration#templating

**Reply by itsnetservices** [reply]
Thanks for your quick reply!

I thought that would be the case. I'm still teaching myself YAML, and I think I'm missing something silly and getting an error of...

**"RoomCardJSTemplateError: SyntaxError: Invalid or unexpected token in 'if (sensor.0x00158d0007ed7959_temperature >= '80') return 'color: red'; if (sensor.0x00158d0007ed7...'"**

Here's my code, could I bother asking you to tell me what I did wrong?

```
info_entities:
  - entity: sensor.0x00158d0007ed7959_temperature
    styles:
      template: >
        if (sensor.0x00158d0007ed7959_temperature >= '80') return 'color: red';
        if (sensor.0x00158d0007ed7959_temperature <= '70') return 'color: blue';
        else return 'color: black';
```

**Reply by itsnetservices** [reply]
NEVERMIND. I figured it out. Sorry! 

Thanks again for your help!

For anyone else with the same issue, here is the working code...
```

info_entities:
  - entity: sensor.0x00158d0007ed7959_temperature
    styles:
      template: >
        if (entity.state >= '80') return 'color: red';
        if (entity.state <= '70') return 'color: blue';
        else return 'color: black';
```

**Comment by undee123** [comment]
Hi,

looking for something similar, but I want to check the state of another entity to change the color, what should I use instaed of "entity.state"?

Thanks for your help!


Edit: if someone is looking for the some, here's how I did it:
`if (states['sensor.mysensorname'].state < 20) return 'color: red';`

**Comment by fwiehl** [comment]
It is not working with single symbols...
I want to change the icon color for

entities:
  - entity: media_player.hifi
    name: Yamaha HiFi
    show_icon: true
    icon: mdi:speaker-wireless
    state_color: true

At the moment the icon is darkblue if deactivated and lightblue if activated.

How can I change the two colors?

**Comment by fwiehl** [comment]
I got it. I works now with this code:

entities:
  - entity: media_player.hifi
    name: Yamaha HiFi
    show_icon: true
    icon:
      conditions:
        - icon: mdi:speaker-wireless
          condition: equals
          value: "on"
          styles:
            color: orange
        - icon: mdi:speaker-off
          condition: equals
          value: "off"
          styles:
            color: grey
